include/pal.h

```diff
@@ -0,0 +1,71 @@
+/*!
+ * \file            pal.h
+ * \date            2025-09-26
+ * \authors         Mario Mazzara [mario.mazzara@eagletrt.it]
+ *                  Mirko Lana [mirko.lana@eagletrt.it]
+ *
+ * \brief           Peripheral Abstraction Layer (PAL) library, suited for all
+ *                  embedded devices.
+ *
+ * \details         This library provides a set of functions and data structures
+ *                  to abstract the communication with peripherals in a device.
+ *                  It's designed to be platform-independent so a driver must be implemented
+ *                  by the user to provide the necessary functionality.
+ *
+ * \warning         The dynamically allocated memory will not be deallocated automatically but has
+ *                  to be freed by using the arena allocator.
+ */
+
+#ifndef PAL_H
+#define PAL_H
+
+#include "ring-buffer.h"
+
+#include <stddef.h>
+#include <stdint.h>
+
+/*!
+ * \brief           Enumeration with all possible return codes of the library.
+ */
+enum PalReturnCode {
+    PAL_RC_OK,            /*!< Everything is fine. */
+    PAL_RC_INVALID_PARAM, /*!< Invalid parameter data. */
+    PAL_RC_NULL_PTR,      /*!< Unexpected NULL pointer. */
+    PAL_RC_IO_ERR,        /*!< Generic I/O error. */
+    PAL_RC_QUEUE_FULL,    /*!< The queue is full. */
+    PAL_RC_QUEUE_EMPTY,   /*!< The queue is empty. */
+    PAL_RC_DESER_ERR      /*!< Deserialization error. */
```

Comment:
Avoid shortening too much identifiers otherwise the code will become difficult to understand and maintain, especially for other people.
```suggestion
    PAL_RC_DESERIALIZATION_ERROR      /*!< Deserialization error. */
```